routes/chatMessageRoutes.js

```diff
@@ -0,0 +1,20 @@
+const express = require("express");
+
+
+// NEW CONTROLLERS
+const Message = require("../controllers/v1/chat_message");
+
+// MIDDLEWARE
+const authMiddleware = require("../middleware/auth");
+
+const router = express.Router();
+
+router.use(authMiddleware.protect);
+
+router.get("/:id", Message.getAllMessages);
+
+router.post("/", Message.createMessage);
```

Comment:
add middleware to upload images, and videos, check out how it is done in module routes!!
in fileUploader file add middlewares for images or videos that might be shared , they should be sent as `message` which is the field you have in the chat schema, when they are uploaded they return a url which is stored as string!! make sure to add this too in your chat_message controller (look on how it's done on the oother controllers that uploads images)